src/jail_run.c

```diff
@@ -246,8 +246,14 @@ jail_run(
             const struct group* gr = getgrnam(group);
 
             if (gr) {
-                egid = gr->gr_gid;
-                GDEBUG("Setting effective GID to %s (%u)", group, (uint)egid);
+                if (egid == 0) {
+                    egid = gr->gr_gid;
+                    GDEBUG("Setting effective GID to %s (%u)", group,
+                        (uint)egid);
+                } else if (egid != gr->gr_gid) {
```

Comment:
Unless I'm mistaken in this stage egid == sailjail->parent_process->egid i.e. whatever it was that executed sailjail.

I do not get the point of this. Permissions are for the launched application. What process requests the launch should not matter and it should be perfectly fine for  all applications to launch any other application e.g. browser with url pointing to application help/home page, or?

Also, does this break dbus autolauch? Are any of the processes related to that that are running as egid=privileged?

Reply:
The idea is to prevent something like `sailjail -p /tmp/foo.profile` where foo.profile contains
```ini
[Sailjail]
Permissions=Privileged
```
from a non-privileged context.